lib/superagent-mock.js

```diff
@@ -142,7 +141,7 @@ function mock (superagent, config, logger) {
       }
     }
 
-    fn(error, response);
+    setTimeout(function () { fn(error, response); }, context.delay || 0);
```

Comment:
You are referring to context while it wasn't declared in that scope (it was declared under the try block scope above). Needs getting it out before the try-catch block.